bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/helper/image/WOIDUrlBuilderHelper.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.sling.api.resource.ValueMap;
+
+import java.awt.*;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WOIDUrlBuilderHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WOIDUrlBuilderHelper.class);
+
+    private static String COMMA = ",";
+    private static String WIDTH_PARAMETER = "width";
+    private static String QUALITY_PARAMETER = "quality";
+    private static String CROP_PARAMETER = "c";
+    private static String ROTATE_PARAMETER = "r";
+    private static String FLIP_PARAMETER = "flip";
+    private static String SIZE_PARAMETER = "sz";
+    private static String FORMAT_PARAMETER = "format";
+    private static String PATH_PARAMETER = "path";
+    private static String SEO_PARAMETER = "seoname";
+
+
+    public static String getSrcSet(@NotNull AssetDelivery assetDelivery, @NotNull Resource assetResource, @NotNull String imageName,
+                                   @NotNull String extension, @NotNull ValueMap componentProperties,
+                                   int[] smartSizes, Dimension originalDimension, int jpegQuality) {
+
+        if (smartSizes.length == 0) {
+            return null;
+        }
+        List<String> srcsetList = new ArrayList<String>();
+        for (int i = 0; i < smartSizes.length; i++) {
+            String src =  getSrc(assetDelivery, assetResource,  imageName, extension, componentProperties, new int[]{smartSizes[i]}, jpegQuality, originalDimension);
+            if (!StringUtils.isEmpty(src)) {
+                srcsetList.add(src);
+            }
+        }
+
+        if (srcsetList.size() > 0) {
+            return StringUtils.join(srcsetList, COMMA);
+        }
+
+        return null;
+    }
+
+    public static  String getSrc(@NotNull AssetDelivery assetDelivery, @NotNull Resource assetResource, @NotNull String imageName,
+                                 @NotNull String extension, @NotNull ValueMap componentProperties,
+                                 @NotNull int[] smartSizes, @NotNull int jpegQuality, Dimension originalDimension) {
+
+        Map<String, Object> params = new HashMap<>();
+
+        String assetPath = componentProperties.get(DownloadResource.PN_REFERENCE, String.class);
+
+        if (StringUtils.isEmpty(imageName) || StringUtils.isEmpty(assetPath)
+                || StringUtils.isEmpty(extension) || "svg".equalsIgnoreCase(extension)) {
+            return null;
+        }
+
+        params.put(PATH_PARAMETER, assetPath);
+        params.put(SEO_PARAMETER, imageName);
+        params.put(FORMAT_PARAMETER, extension);
+
+        if (smartSizes.length == 1) {
```

Comment:
For readability and easier unit-testing I would move following parameter processing to dedicated methods:
* `addSizeParams(params, componentProperties)`
* ...
* `addFlipParams(params, componentProperties)`